src/main/scala/circt/Convention.scala

```diff
@@ -0,0 +1,52 @@
+package chisel3.circt
+
+import chisel3.RawModule
+import chisel3.experimental.{annotate, ChiselAnnotation}
+import firrtl.annotations.{ModuleName, SingleTargetAnnotation}
+import firrtl.FirrtlUserException
+
+/** Annotation to force the compiler to use a specified port convention.
+  * May only be applied to modules.
+  */
+case class ConventionAnnotation(target: ModuleName, convention: String) extends SingleTargetAnnotation[ModuleName] {
```

Comment:
`ModuleTarget` is a slightly better API than `ModuleName` (though they are basically the same thing). This requires using `toTarget` below as opposed to `toNamed` in the Chisel API.